core/src/avm2/e4x.rs

```diff
@@ -470,11 +470,17 @@ impl<'gc> E4XNode<'gc> {
         }
 
         // 5. If Type(V) is XML and V.[[Class]] ∈ {"element", "comment", "processing-instruction", "text"}
-        if let Some(xml) = value.as_object().and_then(|x| x.as_xml_object()) {
-            if matches!(*xml.node().kind(), E4XNodeKind::Attribute(_)) {
-                return Ok(());
-            }
+        if let Some(xml) = value
+            .as_object()
+            .and_then(|x| x.as_xml_object())
+            .and_then(|x| {
```

Comment:
You can use `filter` instead.

Reply:
Thanks. I had no idea that `filter` also existed for `Option`.